src/bot/Bot.ts

```diff
@@ -1,37 +1,106 @@
+import Command from './commands/Command';
+import AddGreetingCommand from './commands/add-greeting.command';
+import CharacterCommand from './commands/character.command';
+import EightBallCommand from './commands/eight-ball.command';
+import HelpCommand from './commands/help.command';
+import PingCommand from './commands/ping.command';
+import SeeCommand from './commands/see.command';
+import QuotesCommand from './commands/quotes.command';
 
 export default class Bot {
   private _discordClient: Discord.Client;
 
-  private _commandList: Discord.Collection<string, ICommand>;
+  private _commandList: Discord.Collection<string, Command>;
 
-  constructor(
-    private _configService: ConfigService,
-    private _loggerService: LoggerService,
-    private _mongoService: MongoService,
-    private _databaseService: DatabaseService,
-  ) {
+  constructor(private _dependencies: Dependencies) {
     this._discordClient = new Discord.Client();
-    this._commandList = new Discord.Collection<string, ICommand>();
+    this._commandList = new Discord.Collection<string, Command>();
+  }
+
+  private setCommands(): void {
+    const prefix = this._dependencies.configService.get('BOT_TRIGGER');
+
+    // Load in our commands for the command handler
+    // TODO: Refactor this into a CommandHandler class?
+    const addGreetingCmd = new AddGreetingCommand(
+      this._dependencies,
+      'addgreeting',
+      ['ag'],
+      'Adds a string to the list greetings used when new users connect to server! Include `{name}` in your message to replace with the new users name.',
+      [`\`${prefix}addgreeting Welcome to the club {name}\``],
+    );
+    const characterCmd = new CharacterCommand(
+      this._dependencies,
+      'character',
+      ['c'],
+      'Adds a string to the list greetings used when new users connect to server! Include `{name}` in your message to replace with the new users name.',
+      [`\`${prefix}addgreeting Welcome to the club {name}\``],
+    );
+    const eightBallCmd = new EightBallCommand(
+      this._dependencies,
+      '8ball',
+      ['eightball', 'magicball', 'ball', 'wisdomball'],
+      'Ask the magic eightball for advice.',
+      [`\`${prefix} 8ball will I be awesome today?\``],
+    );
+    const helpCmd = new HelpCommand(
+      this._dependencies,
+      'help',
+      ['commands'],
+      'Lists available commands and their usage.',
+      [`\`${prefix}help\``, `\`${prefix}help ping\``],
+    );
+    const pingCmd = new PingCommand(
+      this._dependencies,
+      'ping',
+      ['hello'],
+      'Responds, kind of like telling you the bot is alive.',
+      [`\`${prefix}ping\``],
+    );
+    const seeCmd = new SeeCommand(
+      this._dependencies,
+      'see',
+      ['me'],
+      'Sends a DM telling you information about your user on given server.',
+      [`\`${prefix}see\``],
+    );
+    const quoteCmd = new QuotesCommand(
+      this._dependencies,
+      'quotes',
```

Comment:
I'll be first to comment. Not sure if it's better to pass them here or if it would be better to set the values in each class.